src/ui/display_nbgl.c

```diff
@@ -322,347 +127,460 @@ void ui_display_output_address_amount_flow(int index) {
+    pairList.nbPairs = 2;
+    pairList.pairs = pairs;
 
-    nbgl_useCaseReviewStart(&C_Bitcoin_64px,
-                            "Register cosigner",
-                            "",
-                            "Cancel",
-                            continue_light_notify_callback,
-                            ux_flow_response_false);
-}
+    contentList[0].type = CENTERED_INFO;
+    contentList[0].content.centeredInfo.icon = &C_Bitcoin_64px;
+    contentList[0].content.centeredInfo.text1 = "Register cosigner";
+    contentList[0].content.centeredInfo.text2 = NULL;
+    contentList[0].content.centeredInfo.text3 = NULL;
+    contentList[0].content.centeredInfo.style = LARGE_CASE_BOLD_INFO;
+    contentList[0].content.centeredInfo.offsetY = 0;
+    contentList[0].contentActionCallback = NULL;
 
-static void suspicious_pubkey_warning(void) {
-    nbgl_useCaseReviewStart(&C_round_warning_64px,
-                            "WARNING",
-                            "The derivation path\nis unusual",
-                            "Cancel",
-                            continue_light_notify_callback,
-                            ux_flow_response_false);
-}
+    contentList[1].type = TAG_VALUE_LIST;
+    memcpy(&contentList[1].content.tagValueList, &pairList, sizeof(nbgl_layoutTagValueList_t));
+    contentList[1].contentActionCallback = NULL;
 
-void ui_display_pubkey_suspicious_flow(void) {
-    transactionContext.tagValuePair[0].item = "Path";
-    transactionContext.tagValuePair[0].value = g_ui_state.path_and_pubkey.bip32_path_str;
+    contentList[2].type = INFO_BUTTON;
+    contentList[2].content.infoButton.text = "Confirm cosigner";
+    contentList[2].content.infoButton.icon = &C_Bitcoin_64px;
+    contentList[2].content.infoButton.buttonText = "Approve";
+    contentList[2].content.infoButton.buttonToken = REVIEW_CONFIRM;
+    contentList[2].content.infoButton.tuneId = TUNE_TAP_CASUAL;
+    contentList[2].contentActionCallback = generic_content_callback;
 
-    transactionContext.tagValuePair[1].item = "Public key";
-    transactionContext.tagValuePair[1].value = g_ui_state.path_and_pubkey.pubkey;
+    genericContent.callbackCallNeeded = false;
+    genericContent.contentsList = contentList;
+    genericContent.nbContents = 3;
 
-    transactionContext.tagValueList.nbPairs = 2;
+    nbgl_useCaseGenericReview(&genericContent, "Cancel", status_cancel);
+}
 
-    transactionContext.confirm = "Approve public key";
-    transactionContext.confirmed_status = "PUBLIC KEY\nAPPROVED";
-    transactionContext.rejected_status = "Public key rejected";
-    nbgl_useCaseReviewStart(&C_Bitcoin_64px,
-                            "Confirm public key",
-                            "",
-                            "Cancel",
-                            suspicious_pubkey_warning,
-                            status_cancel);
+void ui_display_pubkey_suspicious_flow(void) {
+    confirmed_status = "PUBLIC KEY\nAPPROVED";
+    rejected_status = "Public key rejected";
```

Comment:
To be replaced with nbgl_useCaseReviewStatus calling STATUS_TYPE_ADDRESS_VERIFIED / STATUS_TYPE_ADDRESS_REJECTED?